wallet/src/de/schildbach/wallet/service/work/BaseForegroundWorker.kt

```diff
@@ -23,18 +23,22 @@ abstract class BaseForegroundWorker(
     private val initialTitle: String,
     private val initialContent: String
 ): BaseWorker(context, parameters) {
+    companion object {
+        private val log = LoggerFactory.getLogger(BaseForegroundWorker::class.java)
+    }
     private val notificationManager = NotificationManagerCompat.from(context)
-    abstract suspend fun doWorkInForeground(): Result
+    abstract suspend fun doWorkInForeground(inForeground: Boolean): Result
 
     override suspend fun doWorkWithBaseProgress(): Result {
         return try {
             // Ensure permission is granted before proceeding
             if (hasNotificationPermission()) {
                 setForegroundAsync(createForegroundInfo())
-                doWorkInForeground()
+                doWorkInForeground(true)
             } else {
                 // Log or handle the case where permission is not granted
-                Result.failure()
+                log.info("executing work in background")
+                doWorkInForeground(false)
```

Comment:
allow work to run in the background, but we cannot ensure that it will finish.

If the user did not allow notifications, then wallet restoration previously did not find the identity because the restore identity function was not executed.